### ISSUE TITLE
[Bug]: IntegratedDynamics autocraft causes a deadlock

### ISSUE THREAD

**Issue by libbkmz** [FTBTeam/FTB-Modpack-Issues#512]
### Modpack

FTB OceanBlock

### Modpack version

1.11.0

### Log Files

https://gist.github.com/libbkmz/3820ecd8450ffc11d929b23909d5ebbe - Stack trace captured after getting deadlock via `jstack PID`
https://youtu.be/UwVeBYYL2QA - video with reproduction

### Describe the bug

AutoCraft in Integrated Dynamics creates a deadlock on the server side [at this line](https://github.com/CyclopsMC/IntegratedDynamics/blob/1.16.5-1.10.6/src/main/java/org/cyclops/integrateddynamics/core/network/PositionedAddonsNetworkIngredients.java#L79)
Or [this line](https://github.com/CyclopsMC/IntegratedDynamics/blob/1.16.5-1.10.6/src/main/java/org/cyclops/integrateddynamics/api/network/IPositionedAddonsNetworkIngredients.java#L84)
Also, you can checkout [All stacks ](https://gist.github.com/libbkmz/3820ecd8450ffc11d929b23909d5ebbe) from the JVM. 
Something wrong with this,  when you try to get from CacheStorage:
```
    private final LoadingCache<PartPos, IIngredientComponentStorage<T, M>> cacheStorage;
    ...
    this.cacheStorage = CacheBuilder.newBuilder()
                .build(new CacheLoader<PartPos, IIngredientComponentStorage<T, M>>() {
            public IIngredientComponentStorage<T, M> load(PartPos pos) {
                IIngredientComponentStorage<T, M> storage = getPositionedStorageUnsafe(pos);
                return storage == null ? new IngredientComponentStorageEmpty<>(getComponent()) : storage;
            }
        });
   ...
   this.cacheStorage.get(pos);
```

I was able to reproduce it only in the OceanBlock pack. 

With this mods in list I can't reproduce it:
```IntegratedTunnels 1.8.9
Integrated-Tunnels-Compat 1.8.9
IntegratedTerminals 1.2.11
IntegratedTerminalsCompat 1.2.11
IntegratedDynamics 1.10.6
IntegratedDynamics-Compat 1.10.6  
Integrated NBT
IntegratedCrafting 1.0.20
Cyclops Core 1.12.1
CommonCapabilities 2.8.0
Lollipop 3.2.9
Krate 0.1.4
CreativeCore 2.0.0
JEI 7.7.1.128
Forge 36.2.20
CoFH core 1.4.0.6
JEI Integration 7.0.1.15
TrayLauncher 1.0
CodeChicken Lib 4.0.5.438
ConnectedTexturesMod MC1.16.1-1.1.2.6
CraftingTweaks 12.2.1
```

### Steps to reproduce

I use FTB OceanBlock, I tried to reproduce it with only IntegratedDynamics and Krate and the bug does not occur. Here are the steps to reproduce:
1. Setup 2 chests, 1 crafting table. 
2. Connect chests with tunnels and item interface
3. Connect crafting table with crafting interface
4. Create recipes for "Krates" (again, it shouldn't be related only for them, but haven't tried yet)
5.  Start auto craft for any of them.
6. Sometimes it goes stuck not after the first try, but it could take 1-2-5-10 times before the deadlock.

### Expected behaviour

To see the Screen with Crafting Plan, not the server DeadLock

**Comment by Raidobw2** [comment]
Thanks for the very thorough report. This is going to be very helpful to help the mod author fix it. This is the second report that we got -- the first one fell off and was forgotten sadly.

I'll take that one in charge and relay information to the specific mod authors this time, and interact with them. I've already asked CyclopsCore if I can use the old issue, or create a new one.
If I don't get a reply once business days start again after the Holidays at maximum, I'll have a new issue created for it and linked to here.
Thanks again for the very helpful info!
Basically, this won't fall off this time. :)

**Comment by Raidobw2** [comment]
Adding info. Connected to #346, which was the old issue.

**Comment by Raidobw2** [comment]
Connected to #520 as well.

**Comment by libbkmz** [comment]
What should I do in the meanwhile? I can help to do something, to speed up it.

**Comment by Raidobw2** [comment]
You can interact with the mod author via this issue ^ if you want.
Thanks to your amazing report, they already have a lot of information to work with!

**Comment by Raidobw2** [comment]
Will need to change the config next update.
File `config/integratedterminals-common.toml` and change `craftingPlannerEnableMultithreading` to "false".

Thank you for your help with the issue libbkmz!
